common/ssl_calls.c

```diff
@@ -592,6 +647,15 @@ ssl_tls_accept(struct ssl_tls *self, long ssl_protocols,
                      SSL_MODE_ACCEPT_MOVING_WRITE_BUFFER |
                      SSL_MODE_ENABLE_PARTIAL_WRITE);
     SSL_CTX_set_options(self->ctx, options);
+    DH *dh = get_dh2236 ();
+    if (SSL_CTX_set_tmp_dh(self->ctx, dh) != 1) {
+        g_writeln("SSL_CTX_set_tmp_dh failed");
+        return 1;
+    }
+    DH_free (dh);
```

Comment:
Aren't we freeing `dh` too early?
Did you test it really works with **this particular** dh groups?

Reply:
I didn't checked, have no idea how to and I also didn't checked the internal code for the function itself, I cannot figure out where is it. I just followed the example on the website wiki at https://wiki.openssl.org/index.php/Diffie-Hellman_parameters#Diffie-Hellman